de-lib/src/main/java/org/iplantc/de/pipelines/client/presenter/PipelineViewPresenter.java

```diff
@@ -450,15 +464,12 @@ public void onRemoveAppClicked() {
 
         if (selectedApp != null) {
             ListStore<PipelineTask> store = view.getPipelineAppStore();
-
-            store.remove(selectedApp);
-
-            for (int step = 1; step <= store.size(); step++) {
+            boolean removed = store.remove(selectedApp);
+            for (int step = store.size(); step > 0; step--) {
                 PipelineTask app = store.get(step - 1);
-                app.setStep(step);
+                app.setStep(step - 1);
```

Comment:
So I'm guessing that app steps used to be 1-based, but now they are 0-based? So this is the fix for `CORE-6741 fix step number when removing an app`?

This seems like the only effective change here, since the for-loop looks logically the same to me.
Also, it looks like the `store.remove` call is getting stored in a boolean now, but it's not used?